plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2834,8 +2835,19 @@ else if (column.isHidden()) {
 
     private static void validateColumns(ConnectorTableMetadata tableMetadata)
     {
-        // Validate types are supported
+        // Validate the name and the type of each column
         for (ColumnMetadata column : tableMetadata.getColumns()) {
+            String columnName = column.getName();
+            if (columnName.startsWith(" ")) {
+                throw new TrinoException(INVALID_COLUMN_NAME, format("Column name '%s' is invalid since it contains a leading space", columnName));
```

Comment:
Wording nit: 

```suggestion
                throw new TrinoException(INVALID_COLUMN_NAME, format("Column name '%s' is invalid because it contains a leading space", columnName));
```

Alternatively "Column name '%s' is invalid. It contains a leading space"